scripts/script-tests/resolve-version.test.js

```diff
@@ -0,0 +1,147 @@
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const resolveVersion = require('../resolve-version-lib');
+
```

Comment:
Feels weird to have a `script-tests`. Can we create a module under `tools/resolve-version` instead?

A `resolve-version.sh` can then execute a `yarn build` and call `tools/resolve-version/bin/resolve-version.js`.
If the monorepo is going to complain, we could create another root folder `resolve-version`, not register it as part of the monorepo.

Reply:
That was my original plan but since scripts are running before `yarn install` I didn't want to shift things around too much. I decided to go with `script-tests` private module as a pragmatic way to both allow scripts to continue to function in the same lifecycle stage *and* run tests against them during build.

Reply:
We do something similar here - https://github.com/aws/aws-cdk/blob/master/.github/workflows/pr-linter.yml#L15-L19.
This feels like a better model.

If you include `bump.js` in the module, you can name the module `bump`. And in the future, we could refactor it out into a separate repo.

Reply:
I rather not reorganize where build scripts are sitting in our repo as part of this PR. Happy to consider in a future commit.

Reply:
Ok, that's fine. Please follow up as a separate PR.